src/scripts/quest/classquest/ARC/ClsArc006.cpp

```diff
@@ -0,0 +1,265 @@
+      static constexpr auto Questbattle0 = 2;
+      static constexpr auto Questbattle1 = 3;
+      static constexpr auto Seq0Actor0Lq = 98;
+      static constexpr auto Territorytype0 = 231;
+      static constexpr auto Territorytype1 = 236;
+
+   public:
+      ClsArc006() : Sapphire::ScriptAPI::EventScript( 65612 ){}; 
+      ~ClsArc006(){}; 
+
+   void onTalk( uint32_t eventId, Entity::Player& player, uint64_t actorId ) override
+   {
+      auto pEventMgr = Common::Service< World::Manager::EventMgr >::ref();
+      auto actor = pEventMgr.mapEventActorToRealActor( static_cast<uint32_t>( actorId ) );
+
+      if( actor == Actor0 && !player.hasQuest( getId() ) )
+         Scene00000( player );
+      if( actor == Actor1 && player.getQuestSeq( getId() ) == Seq1 )
+         Scene00002( player );
+      if( actor == Actor0 && player.getQuestSeq( getId() ) == SeqFinish )
+         Scene00013( player );
+      if( actor == Actor7 && player.getQuestSeq( getId() ) == SeqFinish )
+         Scene00014( player );
+      if( actor == Actor8 && player.getQuestSeq( getId() ) == SeqFinish )
+         Scene00015( player );
+      if( actor == Actor9 && player.getQuestSeq( getId() ) == SeqFinish )
+         Scene00016( player );
+      //EOBJECTS
+      else if( actor == Eobject1 )
+         player.eventActionStart( getId(), 0x1B,
+                                  [ & ]( Entity::Player& player, uint32_t eventId, uint64_t additional )
+                                  {
+                                     Scene00011( player );
+                                  },
+                                  nullptr, eventId );
+   }
+
+   private:
+
+   void Scene00000( Entity::Player& player )
+   {
+      player.playScene( getId(), 0, HIDE_HOTBAR,
+                        [ & ]( Entity::Player& player, const Event::SceneResult& result )
+                        {
+                          if( result.param2 == 1 )
+                          {
+                              Scene00001( player );
+                          }
+                        } );
+   }
+
+   void Scene00001( Entity::Player& player )
+   {
+      player.playScene( getId(), 1, SET_EOBJ_BASE | HIDE_HOTBAR | INVIS_EOBJ,
+                        [ & ]( Entity::Player& player, const Event::SceneResult& result )
+                        {
+                          player.updateQuest( getId(), Seq1 );
+                        } );
+   }
+
+   void Scene00002( Entity::Player& player )
+   {
+      player.playScene( getId(), 2, HIDE_HOTBAR,
+                        [ & ]( Entity::Player& player, const Event::SceneResult& result )
+                        {
+                          if( result.param2 == 1 )
+                          {
+                              player.updateQuest( getId(), Seq2 );
+                              // QuestBattle doesn't work
+                              /*auto& pTeriMgr = Common::Service< Sapphire::World::Manager::TerritoryMgr >::ref();
```

Comment:
Getting services from script currently not working if built on windows.

Reply:
I took it from ManFst005 since it worked.
Do I have to wait service fix or is there an alternative way?

Reply:
Code is fine, it's a compiler problem. If you want to test it on Windows then you have to get the service object from a different source (e.g. player) by adding a temp function and call ref() there then return the result to the script.